### PR TITLE
Tour Kit / Welcome Tour: WPCOM Tour Kit variant

### LINKED ISSUE #58643: Tour Kit / Welcome Tour: Extract WPCOM Tour Kit variant
### What

Extract the rest of the Welcome Tour into the tour kit package for reuse. This should unify tour implementations into a single design and skeleton across uses.

### Why

The Welcome Tour contains a few ready parts that we can package together for reuse. Potentially, more tour implementations across WPCOM (and potentially other A8C projects) would benefit from a similar look/feel. Things like the hoverable controls at the top, pagination, previous/next buttons, the overall skeleton of the step view, the minimized view, frame shadows, etc. could be carried over to other tours. 

Ultimately, these boil down to extracting the step and minimized view renderers to someplace convenient. Not related to refactoring the logic for tour definitions and which tour to show.

Screenshot 2021-11-30 at 1 50 29 PM

Screenshot 2021-11-30 at 1 50 36 PM

### How

_No response_

### PR DESCRIPTION
**Pull request by chriskmnds**
### Changes proposed in this Pull Request

Addresses https://github.com/Automattic/wp-calypso/issues/58643

We extract the Welcome Tour step and minimized views into a Tour Kit variant (`WpcomTourKit`). This brings the overlay controls, minimized view, tour rating, pagination controls, all styles, and content skeleton under Tour Kit to unify WPCOM tour implementations. The Welcome Tour becomes an implementation of the variant. Another variant could be a minified view of the same (down the line).

The resulting WPCOM Tour Kit config now omits `renderers` and the Step will ultimately contain the structure:

```
meta: {
	heading: string;
	description: string;
	imgSrc?: {
		src: string;
		type: string;
	};
};
```

Note: Everything is currently divided across `desktop`/`mobile` substructures (something we've planned to refactor/remove).

#### Usage

Same as with the plain Tour Kit component, just without the renderers. Optional tour rating can be enabled for the last step in the tour, along with optional handlers for rating side effects (see the Welcome Tour implementation or Storybook demos, or README).

```
import { WpcomTourKit } from '@automattic/tour-kit';

<WpcomTourKit config={ { ... , tourRating: { enabled: true, ... } } } />
```

### Testing instructions

- Checkout branch, then `yarn dev --sync` from `apps/editing-toolkit` to sync ETK to your sandbox
- Open the Welcome Tour and ensure it runs identical to trunk (no content or structural/style changes)
- Start Storybook `yarn workspace @automattic/tour-kit run storybook` and ensure stories render identical as in trunk
- (once completed) Ensure `WpcomTourKit` stories work both with/without images
- Ensure rating is handled properly and there is no double-rating happening on click (since these send out records to Tracks).

#### Storybook

Kapture 2022-02-22 at 15 26 47

#### Welcome Tour

Kapture 2022-02-22 at 15 34 31

### TODO

- [x] Fix variant styles to accommodate the optional images in the Steps' meta.
- [x] Update README to mention variant.
- [x] Add storybook demos for the WpcomTourKit variant.
- [x] Reintroduce abstracted tour rating logic (#60190).
- [x] Reintroduce pagination control after #60327 is shipped.
- [ ] Improve or remove Error Boundary. [Issue 61842](https://github.com/Automattic/wp-calypso/issues/61842)
- [ ] Fix and add root level CSS class `wpcom-tour-kit`. This was patched with a `__temp__className` prop on Tour Kit, which needs cleanup. Some more general refactor can happen here i.e. the classes we currently pass to Tour Kit are added to the frame, which might be best to add on the root component instead. [Issue 61843](https://github.com/Automattic/wp-calypso/issues/61843)
- [ ] Sanity-check types and resulting render tree (current tree: `WpcomTourKit > TourKit > TourKitStep > WpcomTourKitStep`). [Issue 61844](https://github.com/Automattic/wp-calypso/issues/61844)
- [ ] (follow-up issue) Update types to remove `mobile/desktop` sub structures where relevant (Step CSS classes, Step meta). [Issue 61845](https://github.com/Automattic/wp-calypso/issues/61845)
- [ ] (follow-up issue) Introduce a new translation source as everything is `full-site-editing` still. [Issue 61846](https://github.com/Automattic/wp-calypso/issues/61846)
- [ ] (optional?) "Resume Welcome Tour" in the minimized view is now "Resume Tour" due to this being a more generic component now. We can introduce a tour `title` and reflect that maybe (and ideally communicate that in the README).[ Issue 61847
](https://github.com/Automattic/wp-calypso/issues/61847)

Fixes https://github.com/Automattic/wp-calypso/issues/58643

### CODE REVIEW

**Comment by chriskmnds** [comment]
All TODOs left can be addressed in follow-ups.

File: .yarnrc.yml [hunk 1]
```diff
@@ -278,6 +278,9 @@ packageExtensions:
     peerDependencies:
       react: '*'
       redux: '*'
+  '@wordpress/rich-text@5.1.0':
```
Comment (noahtallen) [inline .yarnrc.yml]: I would avoid adding this, and instead install rich-text@5.0.7. We'll be updating these packages in https://github.com/Automattic/wp-calypso/pull/60161, and we'll make sure all the peer dependency issues get resolved :)

File: packages/tour-kit/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,146 @@
+{
```
Comment (noahtallen) [inline packages/tour-kit/package-lock.json]: Make sure that you're using `yarn install` to update dependencies! We shouldn't have any `package-lock.json` files in `wp-calypso` -- everything will be automatically added to the root `yarn.lock` file when running `yarn install` :)

File: packages/tour-kit/package.json [hunk 3]
```diff
@@ -24,20 +24,28 @@
...
 		"prepack": "yarn run clean && yarn run build",
 		"watch": "tsc --build ./tsconfig.json --watch",
 		"storybook": "start-storybook"
 	},
 	"dependencies": {
+		"@automattic/components": "workspace:^",
+		"@automattic/viewport": "workspace:^",
 		"@automattic/viewport-react": "workspace:^",
 		"@popperjs/core": "^2.10.2",
+		"@wordpress/base-styles": "^4.1.0",
```
Comment (noahtallen) [inline packages/tour-kit/package.json]: I think we are using v4.0.4 in the rest of Calypso, so I would just change these version strings for the `@wordpress` packages to match other package.json files in the repo, and then run `yarn install` again.

**Review by noahtallen — DISMISSED** [review]

**Comment by noahtallen** [comment]
Thanks for those fixes, I removed my review :)

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7172840

Thank you @chriskmnds for including a screenshot in the description! This is really helpful for our translators.

**Comment by chriskmnds** [comment]
Amazing! Thanks for shipping this everyone! @escapemanuele @arcangelini @noahtallen @alshakero  🙇

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7172840).

**Comment by noahtallen** [comment]
> Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7172840).

That took a while 😆

**Pull request merged**